lib/astronoby/angle.rb

```diff
@@ -4,57 +4,97 @@
 
 module Astronoby
   class Angle
-    UNITS = [
-      DEGREES = :degrees,
-      HOURS = :hours,
-      RADIANS = :radians
-    ].freeze
-
-    UNIT_CLASS_NAMES = {
-      DEGREES => "Astronoby::Degree",
-      HOURS => "Astronoby::Hour",
-      RADIANS => "Astronoby::Radian"
-    }
-
     PRECISION = 14
-    PI = BigMath.PI(10)
+    PI = BigMath.PI(PRECISION)
+    FORMATS = %i[dms hms].freeze
 
     class << self
-      UNIT_CLASS_NAMES.each do |unit, class_name|
-        define_method("as_#{unit}") do |angle|
-          Kernel.const_get(class_name).new(angle)
-        end
+      def as_radians(radians)
+        new(radians)
+      end
+
+      def as_degrees(degrees)
+        radians = degrees / BigDecimal("180") * PI
+        new(radians)
+      end
+
+      def as_hours(hours)
+        radians = hours * (PI / BigDecimal("12"))
+        new(radians)
       end
 
       def as_hms(hour, minute, second)
-        angle = hour + minute / 60.0 + second / 3600.0
-        Kernel.const_get(UNIT_CLASS_NAMES[HOURS]).new(angle)
+        hours = hour + minute / 60.0 + second / 3600.0
+        as_hours(hours)
       end
 
       def as_dms(degree, minute, second)
         sign = degree.negative? ? -1 : 1
-        angle = degree.abs + minute / 60.0 + second / 3600.0
-        Kernel.const_get(UNIT_CLASS_NAMES[DEGREES]).new(sign * angle)
+        degrees = degree.abs + minute / 60.0 + second / 3600.0
```

Comment:
Are the 60 and 3600 numbers here the same constants as those from `as_hms`? or are they different constants that happen to have the same values?